REPOSITORY_OVERVIEW.md

```diff
@@ -0,0 +1,1730 @@
+- UART0: USB console (fixed)
+- UART1: TMC steppers or VFD
+- UART2: VFD or Dynamixel or second peripheral
+- Total: 2 configurable UARTs for peripherals
+
+---
+
+## Development Guidelines
+
+### Coding Style
+
+**File**: `CodingStyle.md`
+
+**Key Rules**:
+
+1. **Formatting**: Use `.clang-format` (automatic formatting)
+2. **Naming Conventions**:
+   - Classes/Namespaces: `CamelCase`
+   - Member functions: `snake_case`
+   - Member variables: `_snake_case` (leading underscore)
+3. **File Organization**:
+   - One class per file
+   - Filename matches class name
+   - Header guard: `#pragma once`
+4. **Includes**:
+   - System/library: `<...>`
+   - Local files: `"..."`
+   - CPP file includes its header first
+5. **Namespace Usage**:
+   - No `using namespace` in headers (except in function bodies)
+   - Conservative use in .cpp files
+
+### Adding New Components
+
+#### Adding a New Motor Type
+
+1. Create files: `FluidNC/src/Motors/MyMotor.cpp` and `.h`
+2. Inherit from `MotorDriver` base class
+3. Implement required methods
+4. Register with factory in .cpp file:
+   ```cpp
+   namespace {
+       MotorFactory::InstanceBuilder<MyMotor> registration("my_motor");
+   }
+   ```
+5. Implement `group()` method for configuration
+
+#### Adding a New Spindle Type
+
+1. Create files: `FluidNC/src/Spindles/MySpindle.cpp` and `.h`
+2. Inherit from `Spindle` base class
+3. Implement required methods (init, setState, etc.)
+4. Register with `SpindleFactory`
+
+#### Adding a New Kinematics System
+
+1. Create files: `FluidNC/src/Kinematics/MyKinematics.cpp` and `.h`
+2. Inherit from `Kinematics` base class
+3. Implement transforms: `cartesian_to_motors()` and `motors_to_cartesian()`
+4. Register with `KinematicsFactory`
+
+### Testing
+
+**Unit Tests**: Located in `FluidNC/tests/`
+
+**Test Framework**: Google Test
+
+**Running Tests**:
+```bash
+pio test -e tests
+```
+
+**Fixture Tests**: Located in `fixture_tests/` for integration testing
+
+---
+
+## Installation and Usage
+
+### For End Users
+
+1. **Install Firmware**:
+   - Use installation scripts in `install_scripts/`
+   - Windows: `install_scripts/win64/`
+   - Linux/Mac: `install_scripts/posix/`
+   - Or use pre-built releases from GitHub
+
+2. **Create Configuration**:
+   - Start with example config for your machine type
+   - Modify YAML file for your specific hardware
+   - Validate configuration
+
+3. **Upload Configuration**:
+   - Via USB: Use FluidTerm or other serial terminal
+   - Via WiFi: Use web interface
+   - Command: `$LocalFS/Upload`
+
+4. **Connect and Control**:
+   - **Serial**: USB connection at 115200 baud
+   - **WiFi**: Connect to AP or configure STA mode
+   - **Web UI**: Browse to ESP32's IP address
+   - **GCode Senders**: Compatible with Grbl senders
+
+### FluidTerm Terminal
+
+**Location**: `fluidterm/`
+
+A terminal application specifically designed for FluidNC interaction.
+
+### Configuration Commands
+
+- `$Config/Filename=name.yaml` - Set active config
```

Comment:
This is way too sparse.  It should have a link to the complete information on the wiki